src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioDocumentNavigationService.cs

```diff
@@ -222,20 +246,7 @@ public bool TryNavigateToPosition(Workspace workspace, DocumentId documentId, in
                 var text = document.GetTextSynchronously(CancellationToken.None);
                 var textBuffer = text.Container.GetTextBuffer();
 
-                var boundedPosition = GetPositionWithinDocumentBounds(position, text.Length);
-                if (boundedPosition != position)
-                {
-                    try
-                    {
-                        throw new ArgumentOutOfRangeException();
-                    }
-                    catch (ArgumentOutOfRangeException e) when (FatalError.ReportWithoutCrash(e))
-                    {
-                    }
-                }
-
-                var vsTextSpan = text.GetVsTextSpanForPosition(boundedPosition, virtualSpace);
-
+                var vsTextSpan = getVsTextSpan(text);
```

Comment:
We can't just convert the regular TextSpan to a VsTextSpan because of the virtual case?

Reply:
Only in 1/3 navigation methods do we have a text span, the other two have a line + offset or just position.  All three convert differently to VS text span

Reply:
Ah I see - we also don't always convert to a text span as it's only needed to retrieve the mapped spans.  If the service doesn't exist we skip that check and don't need to ever call the method to retrieve the textspan.